### PR TITLE
New transaction name processor

### PR DESCRIPTION
**Pull request by raphael-theriault-swi**
No need for a cache or anything like that, we can just directly set the attribute on the parent with the new parent storage.

### CODE REVIEW

File: packages/solarwinds-apm/src/processing/transaction-name.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  }
+
+  /** Given a desired transaction name return the one that should be used */
+  registered(name: string): string {
+    const existing = this.#pool.get(name)
+    if (existing !== undefined) {
+      clearTimeout(existing)
+    }
+
+    if (existing !== undefined || this.#pool.size < this.#max) {
```
Comment (tammy-baylis-swi) [inline packages/solarwinds-apm/src/processing/transaction-name.ts:121]: Purpose of this pool makes sense, but I don't think I understand how it works. Is it this?: processor has calculated a name to use (custom or from lambda/http/span name) then checks it against this pool. If the name exists in the pool and we're not yet at pool max, ??? delete after a certain amount of time?

Comment (raphael-theriault-swi) [reply packages/solarwinds-apm/src/processing/transaction-name.ts:121]: Yeah, check if either the pool already had the name or if there's some room we use it, and schedule it to be removed from the pool after the TTL to make some free room. We have to store the scheduled removal tasks themselves in the pool to be able to cancel them when a name is reused.

New name and there is room -> Add the name to pool and schedule to be removed after TTL -> Return name
New name but there is no room -> Return default name
Existing name -> Cancel the removal previously scheduled -> Schedule a new removal -> Return name

Comment (raphael-theriault-swi) [reply packages/solarwinds-apm/src/processing/transaction-name.ts:121]: Adding a comment to clarify that in the implementation !

Comment (tammy-baylis-swi) [reply packages/solarwinds-apm/src/processing/transaction-name.ts:132]: Thank you that's great!

**Review by tammy-baylis-swi — APPROVED** [review]
🔥

**Pull request merged**